core/src/main/java/org/wildfly/channel/ChannelSession.java

```diff
@@ -120,6 +120,29 @@ public MavenArtifact resolveLatestMavenArtifact(String groupId, String artifactI
         return new MavenArtifact(groupId, artifactId, extension, classifier, latestVersion, artifact.file);
     }
 
+    public String findLatestMavenArtifactVersion(String groupId, String artifactId, String extension, String classifier) throws UnresolvedMavenArtifactException {
```

Comment:
Could you use ` findLatestMavenArtifactVersion` inside `resolveLatestMavenArtifact` to avoid code duplication please?